packages/server/src/course/course.controller.ts

```diff
@@ -102,24 +103,49 @@ export class CourseController {
     // }
     const questions = new AsyncQuestionResponse();
     questions.helpedQuestions = all.filter(
-      question => question.status === asyncQuestionStatus.Resolved,
+      (question) => question.status === asyncQuestionStatus.Resolved,
     );
     questions.waitingQuestions = all.filter(
-      question => question.status === asyncQuestionStatus.Waiting,
+      (question) => question.status === asyncQuestionStatus.Waiting,
     );
     questions.otherQuestions = all.filter(
-      question =>
+      (question) =>
         question.status === asyncQuestionStatus.StudentDeleted ||
         question.status === asyncQuestionStatus.TADeleted,
     );
     questions.visibleQuestions = all.filter(
-      question =>
+      (question) =>
         question.visible === true &&
         question.status !== asyncQuestionStatus.TADeleted,
     );
     return questions;
   }
 
+  @Get('limited/:id')
+  async getLimitedCourseResponse(
+    @Param('id') id: number,
+  ): Promise<GetLimitedCourseResponse> {
+    const course = await CourseModel.findOne(id);
```

Comment:
let's check for id and the course code, if there is no restriction on authentication, we don't want to reveal code to public, instead they should provide it to us, and we should check if it's valid.